packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/v1/assistant_wrappers.py

```diff
@@ -0,0 +1,256 @@
+import logging
+import time
+from typing_extensions import override
+from opentelemetry import context as context_api
+from opentelemetry.instrumentation.openai.shared import (
+    _set_span_attribute,
+    model_as_dict,
+)
+from opentelemetry.trace import SpanKind
+from opentelemetry.instrumentation.utils import _SUPPRESS_INSTRUMENTATION_KEY
+
+from opentelemetry.semconv.ai import SpanAttributes, LLMRequestTypeValues
+
+from opentelemetry.instrumentation.openai.utils import _with_tracer_wrapper
+
+logger = logging.getLogger(__name__)
+
+assistants = {}
+runs = {}
```

Comment:
wonder how do these ones behave in an async code

Reply:
I think yes
https://stackoverflow.com/questions/65041691/is-python-dictionary-async-safe